internal/filewriter/filewriter.go

```diff
@@ -0,0 +1,361 @@
+	StateEcefVY     float32 `json:"acoOnGnssStateEcef_vy"`
+	StateEcefVZ     float32 `json:"acoOnGnssStateEcef_vz"`
+	StateEcefX      float32 `json:"acoOnGnssStateEcef_x"`
+	StateEcefY      float32 `json:"acoOnGnssStateEcef_y"`
+	StateEcefZ      float32 `json:"acoOnGnssStateEcef_z"`
+	StateTime       float64 `json:"acoOnGnssStateTime"`
+}
+
+// Records collection of records
+type Records struct {
+	PowerRecords       []PowerRecord       `json:"HK_ecPowOps_1"`
+	CurrentRecords     []CurrentRecord     `json:"HK_scSysOps_1"`
+	TemperatureRecords []TemperatureRecord `json:"HK_tcThermEssential"`
+	AttitudeRecords    []AttitudeRecord    `json:"PreciseAttitudeEstimation"`
+	OrbitRecords       []OrbitRecord       `json:"PreciseOrbitEstimation"`
+	GnssRecords        []GnssRecord        `json:"TM_acGnssOps"`
+}
+
+func getDataset(
+	filename string, groupname string, datasetname string) *hdf5.Dataset {
+
+	f, err := hdf5.OpenFile(filename, hdf5.F_ACC_RDONLY)
+	if err != nil {
+		log.Fatalln(err, filename)
+	}
+	group, err := f.OpenGroup(groupname)
+	if err != nil {
+		log.Fatalln(err, filename, groupname)
+	}
+
+	dataset, err := group.OpenDataset(datasetname)
+	if err != nil {
+		log.Fatalln(err, filename, groupname, datasetname)
+	}
+	return dataset
+}
+
+func getDatasetFloat64(
+	filename string, groupname string, datasetname string) []float64 {
+
+	dataset := getDataset(filename, groupname, datasetname)
+	npoints := dataset.Space().SimpleExtentNPoints()
+	data := make([]float64, npoints)
+	err := dataset.Read(&data)
+	if err != nil {
+		log.Fatalln(err, filename, groupname, datasetname, "data")
+	}
+	dataset.Close()
+	return data
+}
+
+func getDatasetFloat32(
+	filename string, groupname string, datasetname string) []float32 {
+
+	dataset := getDataset(filename, groupname, datasetname)
+	npoints := dataset.Space().SimpleExtentNPoints()
+	data := make([]float32, npoints)
+	err := dataset.Read(&data)
+	if err != nil {
+		log.Fatalln(err, filename, groupname, datasetname, "data")
+	}
+	dataset.Close()
+	return data
+}
+
+func getDatasetUint8(
+	filename string, groupname string, datasetname string) []uint8 {
+
+	dataset := getDataset(filename, groupname, datasetname)
+	npoints := dataset.Space().SimpleExtentNPoints()
+	data := make([]uint8, npoints)
+	err := dataset.Read(&data)
+	if err != nil {
+		log.Fatalln(err, filename, groupname, datasetname, "data")
+	}
+	dataset.Close()
+	return data
+}
+
+func getPowerRecords(filename string) []PowerRecord {
+
+	var group string = "root/Level1A/HK_ecPowOps_1"
+
+	time := getDatasetFloat64(filename, group, "Time")
+	mainBusVoltage := getDatasetFloat32(filename, group, "eciMpduMainBusVoltage")
+	powerheatStr := getDatasetFloat32(filename, group, "ecoUnitPower_heatStr")
+	powerplMain := getDatasetFloat32(filename, group, "ecoUnitPower_plMain")
+	powerplSafe := getDatasetFloat32(filename, group, "ecoUnitPower_plSafe")
+
+	var records []PowerRecord
+	for i := 0; i < len(time); i++ {
```

Comment:
```suggestion
	for i := range time {
```
I think it could be changed to this.

Reply:
fixed

Reply:
fixed